src/uct/tcp/tcp_ep.c

```diff
@@ -646,6 +646,11 @@ static inline ssize_t uct_tcp_ep_sendv(uct_tcp_ep_t *ep)
     return sent_length;
 }
 
+static int uct_tcp_ep_is_conn_closed_by_peer(ucs_status_t io_status)
+{
+    return ((io_status == UCS_ERR_REJECTED) || (io_status == UCS_ERR_TIMED_OUT));
```

Comment:
do we need to check CONNECTION_RESET?
also seems there is redundant ( )

Reply:
yes :(
tests catch this bug...